### PR TITLE
Add client side updates (resolve #147)

### LINKED ISSUE #147: Add client-side updates to Request
It would be nice to be able to give `Request` a callback to refresh the client when a request is made.

Would also be great to have an option to automatically refresh client data in set intervals.

Then we can apply these new features across the instructor center where it makes sense.

## Implementation

- [x] `subscribe` option when `lazy` isn't used, false by default
- [x] Apply subscribe to pieces in instructor center
- [x] request callback option
- [x] Apply request callback option in instructor center

### PR DESCRIPTION
**Pull request by trevordmiller**
# Changes

- Update egghead-ui to get latest `<Request`, which adds `subscribe` (polling) option
- Use `subscribe` in components where data updates regularly
- Lower default page size for pagination as 15 was feeling big; 10 seems more reasonable
- Fix iframe reloading bug with polling by using `shouldComponentUpdate`
- Apply immediate client-side data updates where needed with `Request` callback (`onResponse` handler). This works well for now. It is a bit slower than a server push-based model or client-side only updates with a state management sytem, but it is simpler. If this becomes a problem, I'd recommend we upgrade to GraphQL (most likely with Relay) as it doesn't seem very difficult and has huge wins (maintains caching, requests etc. for you with declarative data fetching syntax inside the component that needs it); this seems to be a much better solution to state and data than Redux or MobX to me if/when the time comes.

# Result For User

client-side-updates

---

### CODE REVIEW

File: src/components/Card/index.js
Comment (trevordmiller) [inline src/components/Card/index.js:11]: Make Card title optional

File: src/components/LessonActions/index.js
Comment (trevordmiller) [inline src/components/LessonActions/index.js:10]: Can update lesson state in an individual lesson or a list of lessons (page)

**Review by joelhooks — APPROVED** [review]

File: src/screens/Lessons/screens/Lesson/components/LessonCard/components/WistiaVideo/index.js
Comment (trevordmiller) [inline src/screens/Lessons/screens/Lesson/components/LessonCard/components/WistiaVideo/index.js:9]: iframes behave oddly, so this ensures no flickering or issues when the wistiaId doesn't change

File: src/components/Card/index.js
Comment (trevordmiller) [reply src/components/Card/index.js:11]: Fixes the prop type error introduced in https://github.com/eggheadio/egghead-instructor-center/pull/166. Not sure I like the title being moved down like this as it is being applied to the video tile in an inconsistent way than the other headings, but this at least fixes the prop type error.

**Comment by trevordmiller** [comment]
~~Waiting on merge of https://github.com/eggheadio/egghead-ui/pull/71 so I can remove yarn link and upgrade `egghead-ui` for real in this PR before merging.~~

**Pull request merged**
